src/core/ad.jl

```diff
@@ -133,18 +133,22 @@ function gradient_logp_reverse(
     model::Model,
     sampler::AbstractSampler=SampleFromPrior(),
 )
-    T = typeof(vi.logp)
+    logp_old = vi.logp
```

Comment:
L136-L151 reversed some recent changes in https://github.com/TuringLang/Turing.jl/pull/857. This could be caused by git merge errors.